sympy/functions/elementary/hyperbolic.py

```diff
@@ -670,10 +670,10 @@ def _eval_as_leading_term(self, x):
         return (1/self._reciprocal_of(self.args[0]))._eval_as_leading_term(x)
 
     def _eval_is_real(self):
-        return self._reciprocal_of(args[0]).is_real
```

Comment:
I think it was not being called.
On Mar 21, 2015 8:46 PM, "Joachim Durchholz" <notifications@github.com>
wrote:

> In sympy/functions/elementary/hyperbolic.py
> <https://github.com/sympy/sympy/pull/9179#discussion_r26892703>:
>
> > @@ -670,10 +670,10 @@ def _eval_as_leading_term(self, x):
> >          return (1/self._reciprocal_of(self.args[0]))._eval_as_leading_term(x)
> >
> >      def _eval_is_real(self):
> > -        return self._reciprocal_of(args[0]).is_real
>
> What got uses for args before the change?
> Just curious - this seems to be a trap for the unwary, and I'd like to see
> how widespread it is.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/9179/files#r26892703>.
>